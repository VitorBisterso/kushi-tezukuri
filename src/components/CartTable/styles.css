.cartTable-container {
  width: 100%;
  border-collapse: collapse;
}

.cartTable-container p {
  margin: 5px 0;
}

.cartTable-container tr:not(:last-of-type) {
  border-bottom: 1px solid var(--COLOR-SHADOW);
}

.cartTable-container td {
  padding: 10px 20px;
}

.cartTable-container td:first-of-type {
  border-right: 1px solid var(--COLOR-SHADOW);
}

.cartTable-container td:nth-child(2):not(:last-of-type) {
  text-align: center;
}

.cartTable-container td:last-of-type {
  border-left: 1px solid var(--COLOR-SHADOW);
  text-align: right;
}

.cartTable-container tr:last-of-type {
  font-family: 'Mali-Bold';
  font-size: var(--TEXT-BIG);
  color: var(--COLOR-GOLD);
}
