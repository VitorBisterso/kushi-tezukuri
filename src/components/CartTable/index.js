import React from 'react'
import PropTypes from 'prop-types'

import Utils from '../../utils'

import './styles.css'

const getTotalPurchase = cart => {
  const initialValue = 0
  return cart.reduce(
    (accumulator, product) => accumulator + product.amount * product.price,
    initialValue
  )
}

const CartTable = ({ cart }) => {
  const totalPurchase = getTotalPurchase(cart)
  const totalPurchaseString = Utils.convertPriceToBrSyntax(totalPurchase)

  const renderProductsLines = () =>
    cart.map(product => {
      const { id, name, price, amount } = product

      const unitPriceString = Utils.convertPriceToBrSyntax(price)
      const totalPriceString = Utils.convertPriceToBrSyntax(price * amount)

      return (
        <tr key={id}>
          <td>
            <p>{name}</p>
          </td>
          <td>
            <p>{`${amount} un. x ${unitPriceString}`}</p>
          </td>
          <td>
            <p>{totalPriceString}</p>
          </td>
        </tr>
      )
    })

  const renderBottomLine = () => (
    <tr>
      <td colSpan={2}>
        <p>Total</p>
      </td>
      <td colSpan={2}>
        <p>{totalPurchaseString}</p>
      </td>
    </tr>
  )

  return (
    <table className="cartTable-container">
      <tbody>
        {renderProductsLines()}
        {renderBottomLine()}
      </tbody>
    </table>
  )
}

const productShape = {
  id: PropTypes.string,
  name: PropTypes.string,
  price: PropTypes.number,
}

CartTable.propTypes = {
  cart: PropTypes.arrayOf(PropTypes.shape(productShape)).isRequired,
}

export default CartTable
