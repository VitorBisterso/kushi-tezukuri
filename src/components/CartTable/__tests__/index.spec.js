import React from 'react'
import { shallow } from 'enzyme'

import CartTable from '../index'

describe('CartTable', () => {
  it('should render correctly', () => {
    const cart = [
      {
        id: '1',
        name: 'MÃ¡scara de bolinhas vermelha e detalhes em cinza',
        price: 6.5,
        typeOfProduct: 'mask',
        typeOfCut: 'F',
      },
    ]
    const component = shallow(<CartTable cart={cart} />)

    expect(component).toMatchSnapshot()
  })
})
