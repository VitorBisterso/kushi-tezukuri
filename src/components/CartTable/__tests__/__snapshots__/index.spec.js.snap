// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartTable should render correctly 1`] = `
<table
  className="cartTable-container"
>
  <tbody>
    <tr
      key="1"
    >
      <td>
        <p>
          MÃ¡scara de bolinhas vermelha e detalhes em cinza
        </p>
      </td>
      <td>
        <p>
          undefined un. x R$ 6,50
        </p>
      </td>
      <td>
        <p>
          R$ NaN,00
        </p>
      </td>
    </tr>
    <tr>
      <td
        colSpan={2}
      >
        <p>
          Total
        </p>
      </td>
      <td
        colSpan={2}
      >
        <p>
          R$ NaN,00
        </p>
      </td>
    </tr>
  </tbody>
</table>
`;
